v4/weaviate/auth/auth.go

```diff
@@ -21,21 +21,22 @@ type Config interface {
 
 type authBase struct{}
 
-func (ab authBase) getIdAndTokenEndpoint(ctx context.Context, con *connection.Connection) (string, string, error) {
+func (ab authBase) getIdAndTokenEndpoint(ctx context.Context, con *connection.Connection) (string, []string, string, error) {
 	rest, err := con.RunREST(ctx, oidcConfigURL, http.MethodGet, nil)
 	if err != nil {
-		return "", "", err
+		return "", []string{}, "", err
 	}
 	cfg := struct {
-		Href     string `json:"href"`
-		ClientID string `json:"clientId"`
+		Href     string   `json:"href"`
+		ClientID string   `json:"clientId"`
+		Scopes   []string `json:"scopes"`
```

Comment:
It would probably be better to define this struct outside the scope of this function, and use it as the return type. It is not very clear what each return type in the signature is supposed to be

Either that, or name the return params

Reply:
went a different route, let me know what you think

Reply:
great, this is also a good solution.